docs/upgrade.md

```diff
@@ -3,5 +3,263 @@ title: Upgrade guide
\ No newline at end of file
+## Migrate from v1 to v2
+
+To make your transition to v2 as smooth as possible while still providing you with the most up-to-date features and patterns we have made targeted breaking changes to some of the utilities. This guide will help you migrate your existing codebase to v2.
+
+### Quick summary
+
+Below a list of the main changes introduced in v2, review the sections below for more details and code examples for the changes that apply to your codebase.
+
+| Area                                   | Change                                                                                                                                           | Code change required |
+| -------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------ | -------------------- |
+| **ES Modules support**                 | Added ES Modules support via dual CommonJS and ES Module bundling, this enables top-level `await` and tree-shaking.                              | -                    |
+| **Middy.js middleware imports**        | Updated import path for Middy.js middlewares to leverage subpath exports - i.e. `@aws-lambda-powertools/tracer/middleware`.                      | Yes                  |
+| **Types imports**                      | Updated import path for TypeScript types to leverage subpath exports - i.e. `@aws-lambda-powertools/logger/types`.                               | Yes                  |
+| **Logger - log sampling**              | Changed implementation of [log sampling](./core/logger.md#sampling-logs) to dynamically switch log level to `DEBUG` on a percentage of requests. | -                    |
+| **Logger - Custom Log Formatter**      | Updated [custom log formatter](#custom-log-formatter) to include additional persistent attributes along with standard structured keys.           | Yes                  |
+| **Logger & Tracer - helper functions** | Removed deprecated `createLogger` and `createTracer` helper functions in favor of direct instantiation.                                          | Yes                  |
+
+### First steps
+
+Before you start, we suggest making a copy of your current working project or create a new git branch.
+
+1. Upgrade Node.js to v16 or higher, Node.js v20 is recommended.
+2. Ensure that you have the latest Powertools for AWS Lambda (TypeScript) version via [Lambda Layer](./index.md#lambda-layer) or npm.
+3. Review the following sections to confirm whether they apply to your codebase.
+
+## ES Modules support
+
+Starting with v2, Powertools for AWS Lambda (TypeScript) supports ES Modules. This means that you can now import the package using the `import` syntax instead of the `require` syntax. This is especially useful if you want to leverage features like top-level `await` in your Lambda function to run asynchronous code during the initialization phase.
```

Comment:
```suggestion
With support for ES Modules in v2, you can now use `import` instead of `require` syntax.

This is especially useful when you want to run asynchronous code during the initialization phase by using top-level `await`.
```

line editing - cut unnecessary words, jumped straight to the action they should be aware, and split the personal touch into an extra line to emphasize its usefulness.

tip: 
- watch out for repeated words with same or similar meanings (`you can now import the package using import`)
- For longer sentences, try using shorter synonyms `to leverage` -> `to use`
- If something isn't immediately obvious when enumerating examples with `features like top-level...`, drop it entirely and focus on one at hand, further improving cognitive load _(working memory)_
- If a new use case is unblocked, lead with that by swapping what you wrote (`This is especially useful when you want to <case>`)